Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -35,14 +35,32 @@ Function Connect-SscServer {
     [Parameter(Mandatory=$true, ParameterSetName='PlainText', Position=1)][string]$username,
     [Parameter(Mandatory=$true, ParameterSetName='PlainText', Position=2)][ValidateNotNullOrEmpty()][string]$password,
     [Parameter(Mandatory=$false, Position=3)][string]$AuthSource='internal',
-    [Parameter(Mandatory=$false, ParameterSetName='Credential')][PSCredential]$Credential
+    [Parameter(Mandatory=$false, ParameterSetName='Credential')][PSCredential]$Credential,
+    [Parameter(Mandatory=$false)][Switch]$SkipCertificateCheck
   )
 
   if ($PSCmdlet.ParameterSetName -eq 'Credential' -AND $Credential -eq $null) { $Credential = Get-Credential}
   if ($Credential) {
     $username = $Credential.GetNetworkCredential().username
     $password = $Credential.GetNetworkCredential().password
   }
+
+  if ($SkipCertificateCheck) {
+    # This if statement is using example code from https://stackoverflow.com/questions/11696944/powershell-v3-invoke-webrequest-https-error
+    add-type @"
+    using System.Net;
+    using System.Security.Cryptography.X509Certificates;
+    public class TrustAllCertsPolicy : ICertificatePolicy {
+        public bool CheckValidationResult(
+            ServicePoint srvPoint, X509Certificate certificate,
+            WebRequest request, int certificateProblem) {
+            return true;
+        }
+    }
+"@
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]'Tls,Tls11,Tls12'

```

Comment:
I'm not sure that this belongs here. Parameter name is SkipCertificateCheck and enabling different security protocols looks like a side effect that is not obvious. I think if you belive this is necessary (I'm not sure about that either) let's make it a separate parameter.

Reply:
I wish this wasn't needed, but from testing with PowerShell 5.1 on Windows 2016, the Connect-SscServer cmdlet is returning `Connect-SscServer : Failure connecting to salt.example.com. The underlying connection was closed: An unexpected error occurred on a receive.`  This issue is described in a couple articles listed below:

- https://devblogs.microsoft.com/powershell/powershell-gallery-tls-support/
- https://www.codyhosterman.com/2016/06/force-the-invoke-restmethod-powershell-cmdlet-to-use-tls-1-2/
- https://www.reddit.com/r/PowerShell/comments/ozr6ye/psa_enabling_tls12_and_you/

Looking at `Invoke-WebRequest` on pwsh 7.2, I see this is implemented through a `-SslProtocol` parameter.  I'll split this out into this parameter in the next commit.